keps/sig-scheduling/3370-native-podgroup-api/README.md

```diff
@@ -0,0 +1,993 @@
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a batch workload user (e.g., MPI/TensorFlow/Spark), I want all tasks of a 
+Job to be either started altogether; otherwise, don’t start any of them.
+
+#### Story 2
+As a service workload user, I want to use PodGroup along with other scheduling
+directives to compose a customized scheduling requirement - like, I want a
+Deployment to run in the same topology (using PodAffinity) at the same time 
+(using PodGroup); otherwise, don’t start any of them.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Most batch jobs consist of multiple (usually two) roles of tasks. For example, a
+typical Tensorflow job has some parameter servers and workers. Each role
+requires a quorum to be ready to run. With that being considered, the following
+API designs all employ a 2-tier specification (Subsets) inside PodGroup. 
+
+The PodGroup API will be defined as a top-level API (in group scheduling.k8s.io, like PriorityClass) 
+
+```go
+// group scheduling.k8s.io
+type PodGroup struct {
+	Spec   PodGroupSpec
+	Status PodGroupStatus
+}
+
+// PodGroupSpec is a description of a PodGroupSpec.
+type PodGroupSpec struct {
+	// Subsets consist of various kinds of pod sets.
+	// A PodGroup is schedulable if all subsets can be scheduled.
+	Subsets []Subset
+
+	// ScheduleTimeoutSeconds defines the timeout threshold to abort
+	// an in-progress PodGroup-level scheduling attempt.
```

Comment:
since when?
What happens when the timeout is reached?

Reply:
> since when?

Calculate the time from the first Pod in this PodGroup to get resources(assume the cache).

> What happens when the timeout is reached?

Free the resources occupied by PodGroup. Avoid long-term resource waste

Reply:
added.